### PR TITLE
Add redux reducers

### PR DESCRIPTION
**Pull request by yosefHesham**
**Brief of what changed:**

Add redux to the app.

---

**Changes Made:**

- Create redux store.
- Created reducers and actions for books.
- Created reducers and actions for categories.

### CODE REVIEW

File: src/redux/books/books.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+export const addBook = (book) => ({
+  type: ADD_BOOK,
+  book,
+});
+
+export const removeBook = (bookId) => ({
+  type: REMOVE_BOOK,
+  bookId,
+
+});
```
Comment (Emmyn5600) [inline src/redux/books/books.js:24]: - [ ] Kindly fix the following issues that have been raised on the console side to make the project run successfully.

image

**Review by Emmyn5600 — CHANGES_REQUESTED** [review]
Hi @yosefHesham 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

 ## Required Changes ♻️
- Check the comment under the review.

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/redux/books/books.js (hunk 1 above)
Comment (yosefHesham) [reply src/redux/books/books.js:24]: @Emmyn5600 
The project does not require any working functionality untill this step.. don't you know?

**Comment by omar-labana** [comment]
# **Approved ✔️**

Hi @yosefHesham

Congratulations on completing this project! 👏🏼 🎉

You have correctly implemented all of the requirements📃 correctly and without any errors⚠️ or bugs🪲.

Cheers and good luck in your next milestones! 🍀

Feel free to leave any questions or comments in the PR thread(don't forget to tag me 😄) if something is not 100% clear.

---
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**